wordpress-mobile/WordPress-FluxC-Android
# Jetpack CP: add support for uploading media using the WPCom's `/wp/v2` API

Jetpack CP sites can't access the regular WPCom APIs that rely on syncing from Jetpack, we need to rely instead on the mirrored `/wp/v2` APIs (more details: p91TBi-67H-p2), this PR adds support for this by adding a new `RestClient` that use this API.

The PR is long because of the tests and the json sample file.

#### Testing
1. You need a site that uses a Jetpack CP plugin (WCPay or Backup) and doesn't have full Jetpack.
2. Open the example app, and login.
3. Click on Media.
4. Select the Jetpack CP site.
5. Test media upload and confirm that it works as expected.

You can also use the WCAndroid's PR for testing: https://github.com/woocommerce/woocommerce-android/pull/5071
